IBAnimatable/Navigator.swift

```diff
@@ -74,3 +75,43 @@ extension Navigator: UINavigationControllerDelegate {
     }
   }
 }
+
+// MARK: - tabbar controller delegate
+extension Navigator: UITabBarControllerDelegate {
+
+  // MARK: - animation controller
+  public func tabBarController(_ tabBarController: UITabBarController,
+                               animationControllerForTransitionFrom fromVC: UIViewController,
+                               to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+
+    interactiveAnimator?.connectGestureRecognizer(to: toVC)
+
+    guard let viewControllers = tabBarController.viewControllers,
+      let fromVCIndex = viewControllers.index(of: fromVC),
+      let toVCIndex = viewControllers.index(of: toVC) else {
+        return nil
+    }
+    if toVCIndex < fromVCIndex {
+      animator?.transitionDuration = transitionDuration
+      return animator
+    } else {
```

Comment:
I can simplify the code here, and I will remove unnecessary else if there is some return